OTAnalytics/domain/geometry.py

```diff
@@ -0,0 +1,78 @@
+from dataclasses import dataclass
+
+from OTAnalytics.domain.common import DataclassValidation
+
+
+@dataclass(frozen=True)
+class Coordinate(DataclassValidation):
+    x: float
+    y: float
+
+
+@dataclass(frozen=True)
+class Line(DataclassValidation):
+    start: Coordinate
+    end: Coordinate
+
+    def _validate(self) -> None:
+        if self.start == self.end:
+            raise ValueError(
+                (
+                    "Start and end point of coordinate must be different to be a line, "
+                    "but are same"
+                )
+            )
+
+
+@dataclass(frozen=True)
+class Polygon(DataclassValidation):
+    coordinates: list[Coordinate]
+
+    def _validate(self) -> None:
+        if len(self.coordinates) < 3:
+            raise ValueError(
+                (
+                    "Number of coordinates to define a valid polygon must be "
+                    f"greater equal three, but is {len(self.coordinates)}"
+                )
+            )
+
+        if self.coordinates[0] != self.coordinates[-1]:
```

Comment:
This means, we need at least 3 coordinates. If exaclty 3 are given, coordinate 1 and must be same. That would be a line. So the number of coordinates must either be greater or equal 4 or coordinates[0] will automatically added at the end to close the polygon.

Reply:
That is correct! Thank you! I'll fix that~